Error happened when make deploy-dev with kustomize v2.1.0

/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

Error happened when make deploy-dev with kustomize v2.1.0

```
(python36) [root@jinchi1 kfserving]# kustomize version
Version: {KustomizeVersion:2.1.0 GitCommit:af67c893d87c5fb8200f8a3edac7fdafd61ec0bd BuildDate:2019-06-18T22:01:59Z GoOs:linux GoArch:amd64}
(python36) [root@jinchi1 kfserving]# make deploy-dev
go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go crd --output-dir=config/default/crds

[debug] name="URL"
[debug] type="string"
[debug] JSONPath=".status.url"
[debug] name="Default Traffic"
[debug] type="integer"
[debug] JSONPath=".status.default.traffic"
[debug] name="Canary Traffic"
[debug] type="integer"
[debug] JSONPath=".status.canary.traffic"
[debug] name="Age"
[debug] type="date"
[debug] JSONPath=".metadata.creationTimestamp"CRD files generated, files can be found under path config/default/crds.
kustomize build config/default/crds -o config/default/crds/serving_v1alpha1_kfservice.yaml
go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go rbac --output-dir=config/default/rbac
RBAC manifests generated under 'config/default/rbac' directory
./hack/image_patch_dev.sh
2019/07/08 01:53:18 Building github.com/kubeflow/kfserving/cmd/manager
2019/07/08 01:53:29 Using base gcr.io/distroless/static:latest for github.com/kubeflow/kfserving/cmd/manager
2019/07/08 01:53:29 No matching credentials were found, falling back on anonymous
2019/07/08 01:53:30 Publishing index.docker.io/jinchi/manager-74c72c260668d14fbfb2548007233c7f:latest
2019/07/08 01:53:32 existing blob: sha256:4003b5b92ca98a8926d9112839f3f17e69f4ec4f995abb188a3ce3ccf93cd6d9
2019/07/08 01:53:32 existing blob: sha256:1558143043601a425aa864511da238799b57fcf7d062d47044f6ddd0e04fe99a
2019/07/08 01:53:33 pushed blob: sha256:2bf05b06970abbf0aeac86cedc810cb828211c97e4e576426aa4fdc31d06f33b
2019/07/08 01:53:51 pushed blob: sha256:c0eb62fb8f661d543bdc572d9abcfcca456cb3ada22cbf586c42ea8a530d4e81
2019/07/08 01:53:52 index.docker.io/jinchi/manager-74c72c260668d14fbfb2548007233c7f:latest: digest: sha256:79a7c26619a027616cbf6e6f5138bb87c33067937a04dead40c950273cca82cb size: 751
2019/07/08 01:53:52 Published index.docker.io/jinchi/manager-74c72c260668d14fbfb2548007233c7f@sha256:79a7c26619a027616cbf6e6f5138bb87c33067937a04dead40c950273cca82cb
kustomize build config/overlays/development | kubectl apply -f -
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe8e851]

goroutine 1 [running]:
sigs.k8s.io/kustomize/k8sdeps/transformer/patch.(*transformer).Transform(0xc00056cea0, 0x12a2ee0, 0xc0003d5de0, 0x7f3045928008, 0x41379a)
	/go/src/sigs.k8s.io/kustomize/k8sdeps/transformer/patch/transformer.go:51 +0x341
sigs.k8s.io/kustomize/pkg/transformers.(*multiTransformer).transform(0xc000562ac0, 0x12a2ee0, 0xc0003d5de0, 0x1, 0x6)
	/go/src/sigs.k8s.io/kustomize/pkg/transformers/multitransformer.go:60 +0x79
sigs.k8s.io/kustomize/pkg/transformers.(*multiTransformer).Transform(0xc000562ac0, 0x12a2ee0, 0xc0003d5de0, 0x6, 0xc000087c20)
	/go/src/sigs.k8s.io/kustomize/pkg/transformers/multitransformer.go:56 +0x84
sigs.k8s.io/kustomize/pkg/accumulator.(*ResAccumulator).Transform(...)
	/go/src/sigs.k8s.io/kustomize/pkg/accumulator/resaccumulator.go:133
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).runTransformers(0xc0003073c0, 0xc000334840, 0x0, 0x0)
	/go/src/sigs.k8s.io/kustomize/pkg/target/kusttarget.go:322 +0x3fb
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).AccumulateTarget(0xc0003073c0, 0xc0000d6660, 0xc0000f00d0, 0xc000425c78)
	/go/src/sigs.k8s.io/kustomize/pkg/target/kusttarget.go:241 +0x2b6
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).makeCustomizedResMap(0xc0003073c0, 0x1, 0xc00000e648, 0x1251e80, 0x1b20538, 0xc000397090)
	/go/src/sigs.k8s.io/kustomize/pkg/target/kusttarget.go:126 +0x2f
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).MakeCustomizedResMap(...)
	/go/src/sigs.k8s.io/kustomize/pkg/target/kusttarget.go:117
sigs.k8s.io/kustomize/pkg/commands/build.(*Options).RunBuild(0xc00038d830, 0x1251d40, 0xc00000e010, 0x127ecc0, 0x1b20538, 0x12947c0, 0x1b20538, 0xc00000e648, 0x1251e80, 0x1b20538, ...)
	/go/src/sigs.k8s.io/kustomize/pkg/commands/build/build.go:130 +0x145
sigs.k8s.io/kustomize/pkg/commands/build.NewCmdBuild.func1(0xc000107b80, 0xc000397220, 0x1, 0x1, 0x0, 0x0)
	/go/src/sigs.k8s.io/kustomize/pkg/commands/build/build.go:80 +0x193
github.com/spf13/cobra.(*Command).execute(0xc000107b80, 0xc0003971e0, 0x1, 0x1, 0xc000107b80, 0xc0003971e0)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.2/command.go:756 +0x465
github.com/spf13/cobra.(*Command).ExecuteC(0xc000106f00, 0xec29d4, 0xc000425f88, 0xc000094058)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.2/command.go:846 +0x2ec
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.2/command.go:794
main.main()
	/go/src/sigs.k8s.io/kustomize/cmd/kustomize/main.go:13 +0x28
error: no objects passed to apply
make: *** [deploy-dev] Error 1
(python36) [root@jinchi1 kfserving]#
```

1. That's OK with kustomize v2.0.3
2. That's OK if executing `make deploy`

**What did you expect to happen:**


**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Istio Version:
- Knative Version: 
- KFServing Version: master
- Kubeflow version: 0.5.1
- Minikube version: 
- Kubernetes version: (use `kubectl version`): v1.11.1+icp
- OS (e.g. from `/etc/os-release`): Redhat 7

--- a/hack/image_patch_dev.sh
+++ b/hack/image_patch_dev.sh
@@ -6,6 +6,7 @@ apiVersion: apps/v1
 kind: StatefulSet 
 metadata:
   name: kfserving-controller-manager
+  namespace: kfserving-system
 spec:
   template:
     spec:

